jni-gen/src/utils.rs

```diff
@@ -82,3 +82,44 @@ pub fn java_identifier_to_rust(name: &str) -> String {
     // identifier can only be composed of [a-zA-Z&_] characters - https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8
     name.replace('_', "__").replace('$', "_dollar_")
 }
+
+pub fn is_signature_of_class_type(signature: &str) -> bool {
+    signature.starts_with('L')
+}
+
+/// Generates a runtime check that the object is of the expected class - applies only to object types (starting with L)
+/// If this value is also the returned value in the setter, unwrap it first (since it is of type Result<T>)
+fn generate_type_check(variable_name: &str, variable_type_name: &str, is_result: bool) -> String {
```

Comment:
Can you add a `debug_assert!(..)` to check that the signature is of a Java class?

In the doc comment, can you update "applies only to object types ..." to clarify that the variable is now *required* to refer to a Java class?